### ISSUE TITLE
Error navigation

### ISSUE THREAD

**Issue by jliptrap** [msuperdock/vim-agda#1]
Is there an easy way to jump to the next error location, such as by populating the quickfix list? 
 Upon an Agda error, I see the filename, line number, and column range in the Agda buffer, but AFAIK it's not in a format that Vim's error navigation understands.  Thanks!

**Comment by jliptrap** [comment]
[vim-fetch](https://github.com/wsdjeg/vim-fetch) would provide a workaround via `gF` if it handled Agda's column format.

**Comment by msuperdock** [comment]
I've not done this for a few reasons (first is the main reason, others are just technical challenges):

- After most errors, Agda stops processing the file and just gives the error. In my experience it's rare that I get enough errors that populating the quickfix list would make sense, and I'm usually happy typing, e.g., `:276` to go to line 276 of my source file.
- The character numbers reported by Agda don't neatly match the character numbers used by vim. (I don't know what the exact issue here is since it's been a while since I wrestled with it, but I think it has to do with widths of Unicode characters.)
- Agda doesn't yet consistently report error & warning ranges in a usable way (though we're working on it!)

An alternative solution that might still meet your needs is that Agda now gives a "JumpToError" message;  we could automatically (maybe optionally) move the cursor to the error location whenever this message is received. Would this meet your needs? If you think you'd really benefit from use of the quickfix list, it would help to give me more context--do you have a use case where you often get a large number of errors that you want to fix without reloading Agda in between?

Example of the "JumpToError" message:

```
module Test where

x

```

Some of the interactive JSON output produced by agda on loading the file:

```
{"kind":"DisplayInfo","info":{"kind":"Error","error":{"message":"/data/code/agda-test/Test.agda:3,1-2\nThe right-hand side can only be omitted if there is an absurd\npattern, () or {}, in the left-hand side.\nwhen checking that the clause x has type _1"},"warnings":[]}}
{"kind":"JumpToError","filepath":"/data/code/agda-test/Test.agda","position":20}
```

(Side note: this "DisplayInfo" message should report the range separately--this is an agda issue.)

**Comment by jliptrap** [comment]
Thanks for the quick and thoughtful reply, your alternative sounds great to me!

**Comment by msuperdock** [comment]
Closed by 89c0ddf.
